### PR TITLE
DMP-1890-download-annotation-document

### PR DESCRIPTION
**Pull request by JasonPittockCGI**
JIRA https://tools.hmcts.net/jira/browse/DMP-1890

Introduce download document functionality

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/resources/openapi/annotations.yaml
Comment (cmagowan-hmcts) [inline src/main/resources/openapi/annotations.yaml:54]: When adding a new endpoint, can you add an example request to the postman collection?

File: src/main/resources/openapi/annotations.yaml
Comment (cmagowan-hmcts) [inline src/main/resources/openapi/annotations.yaml:81]: this example looks like an error?

File: src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java:42]: we should use the annotation contextId here, but its not been merged yet. Tony is introducing it as part of https://github.com/hmcts/darts-api/pull/1032

File: src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java:48]: id's are hardcoded

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:85]: Shoudl probably have the \by\ word in it, i.e. find**By**AnnotationIdAndAnnotationDocumentId

Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:85]: `dirEntity` is not very descriptive, should probably be something like `annotationDocEodEntityOpt`

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:93]: is there a benefit to the judge specific error message?

File: src/main/resources/openapi/annotations.yaml
Comment (JasonPittockCGI) [reply src/main/resources/openapi/annotations.yaml:54]: Sure. Will do.

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:104]: `stream` is quite vague, maybe `blobStream` or something to give a hint as to what it is

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:107]: ExternalFileId is normally blank, i think its only used my ARM. a better field to use might be the eod_id

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:115]: is this not just the annotationDocumentId thats passed in at the top of the method?

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:83]: i don't see any checks that the user downloading the annotation has access to it, it shoudl be somethgin similar to Tonys:-
https://github.com/hmcts/darts-api/pull/1032/files#diff-3cb4a8770477be43d2a61bc42628126ac2d90df6ca290270f9d73f191a6c1ab2R25


or perhaps wait until his is merged and make use of it

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (JasonPittockCGI) [reply src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:85]: Addressed.

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (JasonPittockCGI) [reply src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:104]: Done.

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (JasonPittockCGI) [reply src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:107]: Done

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (JasonPittockCGI) [reply src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:115]: Done

File: src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java
Comment (JasonPittockCGI) [reply src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java:42]: merged Tony's ticket, and used Annotation context_id

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (JasonPittockCGI) [reply src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:93]: This was how I interpreted the ticket. Do u think it incorrect?

File: src/integrationTest/java/uk/gov/hmcts/darts/annotation/AnnotationControllerTest.java
Comment (tonymort) [inline src/integrationTest/java/uk/gov/hmcts/darts/annotation/AnnotationControllerTest.java:55]: Is DirtiesContext neccessary? Slow the test suite down

File: src/integrationTest/java/uk/gov/hmcts/darts/annotation/AnnotationControllerTest.java
Comment (tonymort) [inline src/integrationTest/java/uk/gov/hmcts/darts/annotation/AnnotationControllerTest.java:141]: We dont want to introduce more transactional tests.  What is the reason for it?  If not sure I can take a look.

File: src/main/resources/openapi/annotations.yaml
Comment (cmagowan-hmcts) [inline src/main/resources/openapi/annotations.yaml:189]: the enum is not aligned
image

File: src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/annotation/controller/AnnotationController.java:51]: annotationId is in the path, not the body, so `bodyAuthorisation = true,` is not required

File: src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java:98]: i've double checked with Dave, and there is no difference between Admin and Judge errors. i have updated the ACs to clarify

File: src/integrationTest/java/uk/gov/hmcts/darts/annotation/AnnotationControllerTest.java
Comment (JasonPittockCGI) [reply src/integrationTest/java/uk/gov/hmcts/darts/annotation/AnnotationControllerTest.java:141]: Hi Tony. Detached entity. So ran the test with a transaction. Also had to use dirties context to clear data between  test methods

**Review by cmagowan-hmcts — APPROVED** [review]

**Pull request merged**
